### PR TITLE
Update [Feature:PetSet] e2e tests to [Feature:StatefulSet]

### PR DESCRIPTION
**Pull request by janetkuo**
**What this PR does / why we need it**: Update StatefulSet e2e test feature tags from [Feature:PetSet] to [Feature:StatefulSet]

**Special notes for your reviewer**: Ref test-infra changes https://github.com/kubernetes/test-infra/pull/1151

cc @erictune @foxish @kow3ns @enisoc @kubernetes/sig-apps

**Release note**:

```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37243)

### CODE REVIEW

**Comment by foxish** [comment]
This change would need to go in with a corresponding change in test-infra to test the [StatefulSet] jobs. And while we're at it, we should also stop the alpha Jenkins job from testing StatefulSets.

**Comment by foxish** [comment]
Oh, just noticed the other PR. Thanks!

**Comment by janetkuo** [comment]
@k8s-bot bazel test this

**Comment by janetkuo** [comment]
Maybe we should just remove all [Feature] tag from StatefulSet tests, since it's not alpha anymore?

**Comment by mikedanese** [comment]
@k8s-bot bazel test this

**Comment by janetkuo** [comment]
Per discussion with @bprashanth, updated all Stateful Set e2e tests:

1. Network partition: Removed `Feature` tag, keep `Disruptive` tag, they now runs in `Serial` tests (instead of PetSet feature tests)
2. Clustered apps (MySQL, zookeeper, etc): Updated `[Feature:PetSet]` to `[Feature:StatefulSet]`, it now runs in StatefulSet feature tests (instead of PetSet)
3. Other StatefulSet e2e tests: Removed `Feature` tag, keep `Slow` tag, they now runs in `Slow` tests (instead of PetSet feature tests)

File: test/e2e/petset.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ var _ = framework.KubeDescribe("StatefulSet [Slow] [Feature:PetSet]", func() {
 			ExpectNoError(pst.execInPets(ps, cmd))
 		})
 
-		It("should handle healthy pet restarts during scale [Feature:PetSet]", func() {
+		It("should handle healthy pet restarts during scale", func() {
```
Comment (foxish) [inline test/e2e/petset.go]: References "pet", but I can see other places in this file where that is the case as well. Do we want to change the wording in a follow-up later on?

File: test/e2e/petset.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ const (
 // Time: 25m, slow by design.
```
Comment (foxish) [inline test/e2e/petset.go]: Which of these tests are actually slow? I'm not sure that they take more than 5 mins to run on average.

Comment (janetkuo) [inline test/e2e/petset.go]: I checked the logs on GCE, all stateful set tests (other than disruptive and clustered app ones) took <200s. In this case I'll remove `Slow` tag on them.

File: test/e2e/petset.go (hunk 1 above)
Comment (janetkuo) [inline test/e2e/petset.go]: Yeah I can change this in this PR, in a separate commit

Comment (janetkuo) [inline test/e2e/petset.go]: Changed my mind, this will be a follow-up PR

Comment (janetkuo) [inline test/e2e/petset.go]: See https://github.com/kubernetes/kubernetes/pull/37330

**Pull request merged**